base/random.jl

```diff
@@ -221,13 +221,33 @@ end
 ## srand()
 
 """
-    srand([rng=GLOBAL_RNG], seed) -> rng
-    srand([rng=GLOBAL_RNG]) -> rng
+    srand([rng::AbstractRNG=GLOBAL_RNG,] seed::Integer) -> rng
+    srand([rng::AbstractRNG=GLOBAL_RNG,] seed::Vector{UInt32}) -> rng
+    srand([rng::AbstractRNG=GLOBAL_RNG]) -> rng
 
 Reseed the random number generator. If a `seed` is provided, the RNG will give a
 reproducible sequence of numbers, otherwise Julia will get entropy from the system. For
-`MersenneTwister`, the `seed` may be a non-negative integer or a vector of [`UInt32`](@ref)
-integers. `RandomDevice` does not support seeding.
+[`MersenneTwister`](@ref), the `seed` may be a non-negative integer or a vector of [`UInt32`](@ref)
```

Comment:
According to the [https://docs.julialang.org/en/latest/manual/documentation/](doc), "Do not repeat yourself", this is a bit redundant (except for the precision that the integer must be non-negative). Moreover, I realize that `srand` is not for `MersenneTwister` only: other RNGs may accept other kind of seeds. So I think it's better to keep this docstring as it was, except for the added reflinks. I also started a refactoring at https://github.com/JuliaLang/julia/pull/21909/files, which I should finalize.